### PR TITLE
OnlineDQM : Add Quality Tests for L1T Muons

### PR DESCRIPTION
**Pull request by stahlleiton**
Description:

This PR add to CMSSW a set of new Quality Tests applied on each L1T Muon subsystem, and it includes the following list of features:

- Add Quality Tests for each shifter plot in L1T directory related to BMTF, OMTF, EMTF and UGMT.
- Add Quality Tests for each data/emulator mismatch ratio plot in L1TEMU directory related to BMTF, OMTF, EMTF and UGMT.
- Propagate the new L1T Muon QTs to the L1T Summary Plot (EventInfoClient)
- Creates a new L1TEMU Summary Plot for L1T Emulator and propagate the L1TEMU Muon QTs to it.
- Change each L1T Muon Emulator Client from EDAnalyzer to DQMEDHarvester, in order to make them compatible with the merged PR 18751.

NOTE: Since some of the QTs make use of the 2D Chi2 Quality Test introduce in CMSSW PR 18965, this PR depends on 18965 and will require to be **tested with cmsswPR https://github.com/cms-sw/cmssw/pull/18965** , or otherwise the tests will most likely fail.

@thomreis @bortigno

### CODE REVIEW

File: DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# quality tests for L1T Stage2 BMTF 
+ 
+import FWCore.ParameterSet.Config as cms
+
+l1TStage2BMTFQualityTests = cms.EDAnalyzer("QualityTester",
```
Comment (dmitrijus) [inline DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py]: shouldn't this be a harvester as well?

**Comment by dmitrijus** [comment]
+1

File: DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py (hunk 1 above)
Comment (stahlleiton) [inline DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py]: Currently the QualityTester class is still an edm::EDAnalyzer and was not changed by PR 18751 , so I am not sure if the class should be changed to a DQMEDHarvester.

File: DQM/L1TMonitorClient/python/L1TStage2EmulatorEventInfoClient_cfi.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    #
+    # for each L1 trigger object, give:
+    #     - ObjectLabel:  object label as used in enum L1GtObject
+    #     - ObjectDisable: emulator mask: if 1, the system is masked in the summary plot
+    #
+    # the position in the parameter set gives, in reverse order, the position in the reportSummaryMap
+    # in the trigger object column (right column)
+    L1Objects = cms.VPSet(
+                    cms.PSet(
+                        ObjectLabel = cms.string("TechTrig"),
```
Comment (davidlange6) [inline DQM/L1TMonitorClient/python/L1TStage2EmulatorEventInfoClient_cfi.py]: please try to avoid the duplicated 100s of lines below. Perhaps a loop?
@stahlleiton

File: DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py (hunk 1 above)
Comment (davidlange6) [inline DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py]: is there not a definitive recipe?
@dmitrijus

Comment (dmitrijus) [inline DQM/L1TMonitorClient/python/L1TStage2BMTFQualityTests_cfi.py]: @davidlange6 Not yet. Eventually, QualityTester should be migrated into a harvester, eventually...
But this is out of scope for this PR.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1
